android/app/src/main/kotlin/org/getlantern/lantern/util/PaymentsUtil.kt

```diff
@@ -244,12 +238,46 @@ class PaymentsUtil(private val activity: Activity) {
                         return
                     }
 
+
+                    if (purchases[0].purchaseState != Purchase.PurchaseState.PURCHASED) {
+                        /*
+                        * if the purchase state is not purchased then do not call api
+                        * make user pro temporary next user open app it will check the purchase state and call api accordingly
+                        * */
+                        session.linkDevice()
+                        lanternClient.userData(object : LanternHttpClient.ProUserCallback {
+                            override fun onSuccess(response: Response, userData: ProUser) {
+                                Logger.e(TAG, "User detail : $userData")
+                                session.setIsProUser(true)
+                                activity.runOnUiThread {
+                                    methodCallResult.success("purchaseSuccessful")
+                                }
+                            }
+
+                            override fun onFailure(throwable: Throwable?, error: ProError?) {
+                                Logger.error(TAG, "Unable to fetch user data: $throwable.message")
+                                activity.runOnUiThread {
+                                    methodCallResult.success("purchaseSuccessful")
```

Comment:
Just checking, do we want to call `methodCallResult.success` if there's an error fetching user data?

Reply:
I had the same thought on this, But then I found out that the server doesn't make sure pro is the right way and takes some time, So if the purchase is a success we should use a pro, If something is wrong we are already calling API on next time user start app, @atavism  What do you think on that?